### PR TITLE
Configure workers in Async worker pool

### PR DESCRIPTION
**Pull request by pxp9**

### CODE REVIEW

File: src/asynk/async_worker_pool.rs [hunk 1]
```diff
@@ -51,8 +47,9 @@ where
     }
 
     #[async_recursion]
-    pub async fn supervise_worker(queue: AsyncQueue<Tls>) -> Result<(), Error> {
-        let result = Self::run_worker(queue.clone()).await;
+    pub async fn supervise_worker(queue: AQueue) -> Result<(), Error> {
+        let q: &'static mut AQueue = &mut queue.clone();
```
Comment (pxp9) [inline src/asynk/async_worker_pool.rs:51]: we have a borrowing error here , but if we solve them i think this will be the way to generic this

Comment (pxp9) [reply src/asynk/async_worker_pool.rs]: we will think later

File: fang_examples/simple_async_worker/src/lib.rs [hunk 2]
```diff
@@ -26,8 +26,10 @@ impl AsyncRunnable for MyTask {
         tokio::time::sleep(Duration::from_secs(3)).await;
 
         let new_task = MyTask::new(self.number + 1);
-        let metadata = serde_json::to_value(&new_task as &dyn AsyncRunnable).unwrap();
-        queue.insert_task(metadata, "common").await.unwrap();
+        queue
```
Comment (ayrat555) [inline fang_examples/simple_async_worker/src/lib.rs:29]: please rebase the pr. it seems changes related to `insert_task` were already merged

File: src/asynk/async_queue.rs [hunk 3]
```diff
@@ -95,6 +96,8 @@ pub enum AsyncQueueError {
     PoolError(#[from] RunError<bb8_postgres::tokio_postgres::Error>),
     #[error(transparent)]
     PgError(#[from] bb8_postgres::tokio_postgres::Error),
+    #[error(transparent)]
+    SerdeError(#[from] serde_json::Error),
```
Comment (ayrat555) [inline src/asynk/async_queue.rs:100]: this change was also merged

**Review by ayrat555 — APPROVED** [review]
looks good. just rebase

File: src/asynk/async_queue.rs (hunk 3 above)
Comment (pxp9) [reply src/asynk/async_queue.rs]: rebased

File: fang_examples/simple_async_worker/src/lib.rs (hunk 2 above)
Comment (pxp9) [reply fang_examples/simple_async_worker/src/lib.rs]: rebased

**Review by ayrat555 — APPROVED** [review]

**Comment by pxp9** [comment]
Solve #50

**Pull request merged**